packages/sync-actions/test/product-sync-variants.spec.js

```diff
@@ -291,42 +291,186 @@ describe('Actions', () => {
     ])
   })
 
-  it('should handle mapping actions for new variants without masterVariant',
-  () => {
-    const before = {
-      id: '123',
-      version: 1,
-      masterVariant: {
-        id: 1,
-        sku: 'v1',
-        attributes: [{ name: 'foo', value: 'bar' }],
-      },
-      variants: [
-        { id: 2, sku: 'v2', key: 'v2', attributes: [{ name: 'foo', value: 'qux' }] },
-        { id: 3, sku: 'v3', key: 'v3', attributes: [{ name: 'foo', value: 'baz' }] },
-      ],
-    }
+  describe('without master variant in `now`', () => {
+    describe('with master variant in `before`', () => {
+      const before = {
+        id: '123',
+        version: 1,
+        masterVariant: {
+          id: 1,
+          sku: 'v1',
+          attributes: [{ name: 'foo', value: 'bar' }],
+        },
+        variants: [],
+      }
+
+      const now = {
+        id: '123',
+        // <-- no masterVariant
+        variants: [],
+      }
+
+      it('should not generate update action to remove master variant',
```

Comment:
Building from comment above, this test fails

```
describe('no mastervariant in `now` obj', () => {
  const before = {
    id: '123',
    version: 1,
    masterVariant: {
      id: 1,
      sku: 'v1',
      attributes: [{ name: 'foo', value: 'bar' }],
    },
    variants: [],
  }

  const now = {
    id: '123',
    // <-- no masterVariant
    variants: [
      {
        id: 2,
        sku: 'v1',
        attributes: [{ name: 'foo', value: 'bar' }],
      },
    ],
  }

  it('should generate a addVariant action', () => {
    const actions = productsSync.buildActions(now, before)
    expect(actions).toEqual([
      { action: 'addVariant', sku: 'v1', attributes: [{ name: 'foo', value: 'bar' }] },
    ])
  })
})
```

Reply:
Yeah, cause that test lives in the grey zone mentioned above. What do we assume to be intended here? You had a master variant and now you don't want one. Which basically is breaking with our invariants. 

Moreover, the test fails as the `id` 1 doesn't exist anymore. 

1. So should it be removed? (`1`)?
2. Should `2` be implicitly be the new master?
    Did use user intend to do that?
3. Should be keep both?

I think with `now` you declare where you want to add up at. This test is flawed cause the state the user wants to end up in is not valid and we can only assume what he meant.